Gems/AtomTressFX/Code/Rendering/HairFeatureProcessor.cpp

```diff
@@ -316,6 +316,62 @@ namespace AZ
                 return pass ? true : false;
             }
 
+            bool HairFeatureProcessor::CreateHairParentPass(RPI::RenderPipeline* renderPipeline)
+            {
+                if (HasHairParentPass(renderPipeline))
+                {
+                    CreatePerPassResources();
+                    return false;
+                }
+                const char* passRequestAssetFilePath = "Passes/AtomTressFX_PassRequest.azasset";
+                m_hairPassRequestAsset = AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::AnyAsset>(
+                    passRequestAssetFilePath, AZ::RPI::AssetUtils::TraceLevel::Warning);
+                const AZ::RPI::PassRequest* passRequest = nullptr;
+                if (m_hairPassRequestAsset->IsReady())
+                {
+                    passRequest = m_hairPassRequestAsset->GetDataAs<AZ::RPI::PassRequest>();
+                }
+                if (!passRequest)
+                {
+                    AZ_Warning("AtomTressFx", false, "Failed to load PassRequest from %s", passRequestAssetFilePath);
+                    return false;
+                }
+
+                m_usePPLLRenderTechnique = passRequest->m_templateName == AZ::Name("HairParentPassTemplate");
+
+                 // Create the pass
+                RPI::Ptr<RPI::Pass> hairParentPass  = RPI::PassSystemInterface::Get()->CreatePassFromRequest(passRequest);
+
+                // add the pass to render pipeline
+                RPI::Ptr<RPI::ParentPass> rootPass = renderPipeline->GetRootPass();
+
+                // Find opaque pass
+                auto passFilter = RPI::PassFilter::CreateWithTemplateName(Name("OpaqueParentTemplate"), renderPipeline);
+                RPI::Ptr<RPI::Pass> opaquePass = nullptr;
+                RPI::PassSystemInterface::Get()->ForEachPass(passFilter, [&opaquePass](RPI::Pass* pass) -> RPI::PassFilterExecutionFlow
+                    {
+                        opaquePass = pass;
+                        return RPI::PassFilterExecutionFlow::StopVisitingPasses;
+                    });
+
+                if (!opaquePass)
+                {
+                    return false;
```

Comment:
This and all other false returns should have an error message that hair parent was not created and hair will not be rendered due to ..

Reply:
I didn't treat it as an error since we won't want to insert hair pass for some render pipeline if they are not supposed to have hair. 
I would prefer adding some warning information for each false condition. How to you think?